recipes/fast-dds/all/conanfile.py

```diff
@@ -52,88 +55,87 @@ def _minimum_compilers_version(self):
         }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("tinyxml2/9.0.0")
         self.requires("asio/1.21.0")
-        self.requires("fast-cdr/1.0.23")
-        self.requires("foonathan-memory/0.7.1")
+        self.requires("fast-cdr/1.0.26", transitive_headers=True)
+        self.requires("foonathan-memory/0.7.3")
         self.requires("boost/1.75.0")  # boost/1.76 is required by version 2.3.2, boost/1.75.0 required for 2.3.3 by Windows
         if self.options.with_ssl:
-            self.requires("openssl/1.1.1m")
+            self.requires("openssl/1.1.1t")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if min_version and tools.Version(self.settings.compiler.version) < min_version:
-            raise ConanInvalidConfiguration(
-                "{} requires C++{} support. {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard,
-                    self.settings.compiler, self.settings.compiler.version
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
                 )
-            )
-        if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
-            # This combination leads to an fast-dds error when linking
-            # linking dynamic '*.dll' and static MT runtime
-            raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```

Comment:
Why does this not work? seems odd it used to 🤔

Reply:
fixed. I forgot to make changes after copying from the template.